xbmc/cdrip/EncoderFFmpeg.cpp

```diff
@@ -22,227 +16,233 @@
-
-  if (m_NeedConversion)
+bool CEncoderFFmpeg::Init()
+{
+  try
   {
-    m_SwrCtx = swr_alloc_set_opts(nullptr, m_CodecCtx->channel_layout, m_OutFormat,
-                                  m_CodecCtx->sample_rate, m_CodecCtx->channel_layout, m_InFormat,
-                                  m_CodecCtx->sample_rate, 0, nullptr);
-    if (!m_SwrCtx || swr_init(m_SwrCtx) < 0)
+    ADDON::AddonPtr addon;
+    const std::string addonId = CServiceBroker::GetSettingsComponent()->GetSettings()->GetString(
+        CSettings::SETTING_AUDIOCDS_ENCODER);
+    bool success = CServiceBroker::GetAddonMgr().GetAddon(addonId, addon, ADDON::ADDON_UNKNOWN,
+                                                          ADDON::OnlyEnabled::YES);
+    int bitrate;
+    if (success && addon)
     {
-      CLog::Log(LOGERROR, "CEncoderFFmpeg::{} - Failed to initialize the resampler", __func__);
-      av_frame_free(&m_BufferFrame);
-      av_freep(&m_Buffer);
-      av_freep(&m_Stream);
-      av_freep(&m_Format->pb);
-      av_freep(&m_Format);
-      return false;
+      addon->GetSettingInt("bitrate", bitrate);
+      bitrate *= 1000; /* Multiply as on settings as kbps */
     }
-
-    m_ResampledBufferSize =
-        av_samples_get_buffer_size(nullptr, m_iInChannels, m_NeededFrames, m_OutFormat, 0);
-    m_ResampledBuffer = (uint8_t*)av_malloc(m_ResampledBufferSize);
-    m_ResampledFrame = av_frame_alloc();
-    if (!m_ResampledBuffer || !m_ResampledFrame)
+    else
     {
-      CLog::Log(LOGERROR, "CEncoderFFmpeg::{} - Failed to allocate a frame for resampling",
-                __func__);
-      av_frame_free(&m_ResampledFrame);
-      av_freep(&m_ResampledBuffer);
-      swr_free(&m_SwrCtx);
-      av_frame_free(&m_BufferFrame);
-      av_freep(&m_Buffer);
-      av_freep(&m_Stream);
-      av_freep(&m_Format->pb);
-      av_freep(&m_Format);
-      return false;
+      throw EncoderException("Could not get add-on: {}", addonId);
     }
-    m_ResampledFrame->nb_samples = m_NeededFrames;
-    m_ResampledFrame->format = m_OutFormat;
-    m_ResampledFrame->channel_layout = m_CodecCtx->channel_layout;
-    avcodec_fill_audio_frame(m_ResampledFrame, m_iInChannels, m_OutFormat, m_ResampledBuffer,
-                             m_ResampledBufferSize, 0);
-  }
 
-  /* set the tags */
-  SetTag("album", m_strAlbum);
-  SetTag("album_artist", m_strArtist);
-  SetTag("genre", m_strGenre);
-  SetTag("title", m_strTitle);
-  SetTag("track", m_strTrack);
-  SetTag("encoder", CSysInfo::GetAppName() + " FFmpeg Encoder");
+    // Hack fix about PTS on generated files.
+    // - AAC need to multiply with sample rate
+    //   - Note: Within Kodi it can still played without use of sample rate, only becomes by VLC the problem visible,
+    // - WMA need only the multiply with 1000
+    if (addonId == "audioencoder.kodi.builtin.aac")
+      m_samplesCountMultiply = m_iInSampleRate;
+    else if (addonId == "audioencoder.kodi.builtin.wma")
+      m_samplesCountMultiply = 1000;
```

Comment:
This parts are added new.
Unfortunately, ffmpeg needs two different ways of setting it. This now sets the multiply at init.

Crazy enough, Kodi also works without it and so I didn't notice it, just had a quick test using VLC and then saw that the length was divided by 48.